### ISSUE TITLE
In some condition use sub-query make doris return wrong value

### ISSUE THREAD

**Issue by qdore** [apache/incubator-doris#1097]
**Describe the bug**
1. correct result:
![image](https://user-images.githubusercontent.com/6831829/57220531-3f0af500-702e-11e9-973d-e8c99543d3a9.png)
2. wrong result
![image](https://user-images.githubusercontent.com/6831829/57220547-4fbb6b00-702e-11e9-8d13-f497659019a7.png)

**To Reproduce**
1. Use a normal query sql: 

```
SELECT a.event_day,
          e.new_read_num as new_read_num
   FROM
     (SELECT event_day,
             HLL_UNION_AGG(cuid_uv_set) AS 'all_user_num'
      FROM visd_dws_topic_novel_wide_table_user
      WHERE sid IN ('all')
        AND value IN ('all')
        AND event_day >= STR_TO_DATE('2019-04-29 00:00:00', '%Y-%m-%d %H:%i:%s')
        AND event_day <= STR_TO_DATE('2019-05-06 00:00:00', '%Y-%m-%d %H:%i:%s')
        AND is_dau=1
      GROUP BY event_day) a
     
     LEFT OUTER JOIN
     
     (SELECT t1.event_day,
             count(distinct t1.cuid) AS 'new_read_num'
      FROM
        ( SELECT distinct event_day,
                          cuid
         FROM visd_dws_topic_novel_wide_table_user
         WHERE sid IN ('all')
           AND value IN ('all')
           AND event_day >= STR_TO_DATE('2019-04-29 00:00:00', '%Y-%m-%d %H:%i:%s')
           AND event_day <= STR_TO_DATE('2019-05-06 00:00:00', '%Y-%m-%d %H:%i:%s')
           AND is_read_dau=1 ) t1
      JOIN
        ( SELECT distinct event_day,
                          cuid
         FROM visd_dws_topic_novel_wide_table_user
         WHERE sid IN ('all')
           AND value IN ('all')
           AND event_day >= STR_TO_DATE('2019-04-29 00:00:00', '%Y-%m-%d %H:%i:%s')
           AND event_day <= STR_TO_DATE('2019-05-06 00:00:00', '%Y-%m-%d %H:%i:%s')
           AND is_new=1 ) t2 on t1.event_day = t2.event_day
      and t1.cuid = t2.cuid
      GROUP BY event_day) e ON a.event_day=e.event_day
```

2. move this sql to a sub query [expr_qry]

```
SELECT DATE(event_day) AS __timestamp,
       expr_qry.new_read_num AS new_read_num
FROM
  (SELECT a.event_day,
          e.new_read_num as new_read_num
   FROM
     (SELECT event_day,
             HLL_UNION_AGG(cuid_uv_set) AS 'all_user_num'
      FROM visd_dws_topic_novel_wide_table_user
      WHERE sid IN ('all')
        AND value IN ('all')
        AND event_day >= STR_TO_DATE('2019-04-29 00:00:00', '%Y-%m-%d %H:%i:%s')
        AND event_day <= STR_TO_DATE('2019-05-06 00:00:00', '%Y-%m-%d %H:%i:%s')
        AND is_dau=1
      GROUP BY event_day) a
     
     LEFT OUTER JOIN
     
     (SELECT t1.event_day,
             count(distinct t1.cuid) AS 'new_read_num'
      FROM
        ( SELECT distinct event_day,
                          cuid
         FROM visd_dws_topic_novel_wide_table_user
         WHERE sid IN ('all')
           AND value IN ('all')
           AND event_day >= STR_TO_DATE('2019-04-29 00:00:00', '%Y-%m-%d %H:%i:%s')
           AND event_day <= STR_TO_DATE('2019-05-06 00:00:00', '%Y-%m-%d %H:%i:%s')
           AND is_read_dau=1 ) t1
      JOIN
        ( SELECT distinct event_day,
                          cuid
         FROM visd_dws_topic_novel_wide_table_user
         WHERE sid IN ('all')
           AND value IN ('all')
           AND event_day >= STR_TO_DATE('2019-04-29 00:00:00', '%Y-%m-%d %H:%i:%s')
           AND event_day <= STR_TO_DATE('2019-05-06 00:00:00', '%Y-%m-%d %H:%i:%s')
           AND is_new=1 ) t2 on t1.event_day = t2.event_day
      and t1.cuid = t2.cuid
      GROUP BY event_day) e ON a.event_day=e.event_day) expr_qry
ORDER BY event_day DESC
LIMIT 1000;

```

**Comment by imay** [comment]
@qdore 

1. can you explain these two queries and paste the results?
2. can you try remove `ORDER BY event_day DESC` in query 2 and check the result?

**Comment by qdore** [comment]
1. explain result
(1) right result:
```
| PLAN FRAGMENT 0                                                                                                                                                                                                            |
|  OUTPUT EXPRS:<slot 5> | <slot 30> |                                                                                                                                                                                       |
|   PARTITION: UNPARTITIONED                                                                                                                                                                                                 |
|                                                                                                                                                                                                                            |
|   RESULT SINK                                                                                                                                                                                                              |
|                                                                                                                                                                                                                            |
|   20:EXCHANGE                                                                                                                                                                                                              |
|      tuple ids: 1 10N                                                                                                                                                                                                      |
|                                                                                                                                                                                                                            |
| PLAN FRAGMENT 1                                                                                                                                                                                                            |
|  OUTPUT EXPRS:                                                                                                                                                                                                             |
|   PARTITION: HASH_PARTITIONED: <slot 5>                                                                                                                                                                                    |
|                                                                                                                                                                                                                            |
|   STREAM DATA SINK                                                                                                                                                                                                         |
|     EXCHANGE ID: 20                                                                                                                                                                                                        |
|     UNPARTITIONED                                                                                                                                                                                                          |
|                                                                                                                                                                                                                            |
|   9:HASH JOIN                                                                                                                                                                                                              |
|   |  join op: LEFT OUTER JOIN (BROADCAST)                                                                                                                                                                                  |
|   |  hash predicates:                                                                                                                                                                                                      |
|   |    <slot 5> = <slot 29>                                                                                                                                                                                                |
|   |  tuple ids: 1 10N                                                                                                                                                                                                      |
|   |                                                                                                                                                                                                                        |
|   |----19:EXCHANGE                                                                                                                                                                                                         |
|   |       tuple ids: 10                                                                                                                                                                                                    |
|   |                                                                                                                                                                                                                        |
|   11:AGGREGATE (merge finalize)                                                                                                                                                                                            |
|   |  group by: <slot 5>                                                                                                                                                                                                    |
|   |  tuple ids: 1                                                                                                                                                                                                          |
|   |                                                                                                                                                                                                                        |
|   10:EXCHANGE                                                                                                                                                                                                              |
|      tuple ids: 1                                                                                                                                                                                                          |
|                                                                                                                                                                                                                            |
| PLAN FRAGMENT 2                                                                                                                                                                                                            |
|  OUTPUT EXPRS:                                                                                                                                                                                                             |
|   PARTITION: HASH_PARTITIONED: <slot 27>                                                                                                                                                                                   |
|                                                                                                                                                                                                                            |
|   STREAM DATA SINK                                                                                                                                                                                                         |
|     EXCHANGE ID: 19                                                                                                                                                                                                        |
|     UNPARTITIONED                                                                                                                                                                                                          |
|                                                                                                                                                                                                                            |
|   8:AGGREGATE (update finalize)                                                                                                                                                                                            |
|   |  output: count(<slot 28>)                                                                                                                                                                                              |
|   |  group by: <slot 27>                                                                                                                                                                                                   |
|   |  tuple ids: 10                                                                                                                                                                                                         |
|   |                                                                                                                                                                                                                        |
|   18:AGGREGATE (merge serialize)                                                                                                                                                                                           |
|   |  group by: <slot 27>, <slot 28>                                                                                                                                                                                        |
|   |  tuple ids: 9                                                                                                                                                                                                          |
|   |                                                                                                                                                                                                                        |
|   17:EXCHANGE                                                                                                                                                                                                              |
|      tuple ids: 9                                                                                                                                                                                                          |
|                                                                                                                                                                                                                            |
| PLAN FRAGMENT 3                                                                                                                                                                                                            |
|  OUTPUT EXPRS:                                                                                                                                                                                                             |
|   PARTITION: HASH_PARTITIONED: <slot 14>, <slot 15>                                                                                                                                                                        |
|                                                                                                                                                                                                                            |
|   STREAM DATA SINK                                                                                                                                                                                                         |
|     EXCHANGE ID: 17                                                                                                                                                                                                        |
|     HASH_PARTITIONED: <slot 27>                                                                                                                                                                                            |
|                                                                                                                                                                                                                            |
|   7:AGGREGATE (update serialize)                                                                                                                                                                                           |
|   |  STREAMING                                                                                                                                                                                                             |
|   |  group by: <slot 14>, <slot 15>                                                                                                                                                                                        |
|   |  tuple ids: 9                                                                                                                                                                                                          |
|   |                                                                                                                                                                                                                        |
|   6:HASH JOIN                                                                                                                                                                                                              |
|   |  join op: INNER JOIN (BROADCAST)                                                                                                                                                                                       |
|   |  hash predicates:                                                                                                                                                                                                      |
|   |    <slot 14> = <slot 23>                                                                                                                                                                                               |
|   |    <slot 15> = <slot 24>                                                                                                                                                                                               |
|   |  tuple ids: 4 7                                                                                                                                                                                                        |
|   |                                                                                                                                                                                                                        |
|   |----16:EXCHANGE                                                                                                                                                                                                         |
|   |       tuple ids: 7                                                                                                                                                                                                     |
|   |                                                                                                                                                                                                                        |
|   13:AGGREGATE (merge finalize)                                                                                                                                                                                            |
|   |  group by: <slot 14>, <slot 15>                                                                                                                                                                                        |
|   |  tuple ids: 4                                                                                                                                                                                                          |
|   |                                                                                                                                                                                                                        |
|   12:EXCHANGE                                                                                                                                                                                                              |
|      tuple ids: 4                                                                                                                                                                                                          |
|                                                                                                                                                                                                                            |
| PLAN FRAGMENT 4                                                                                                                                                                                                            |
|  OUTPUT EXPRS:                                                                                                                                                                                                             |
|   PARTITION: HASH_PARTITIONED: <slot 23>, <slot 24>                                                                                                                                                                        |
|                                                                                                                                                                                                                            |
|   STREAM DATA SINK                                                                                                                                                                                                         |
|     EXCHANGE ID: 16                                                                                                                                                                                                        |
|     UNPARTITIONED                                                                                                                                                                                                          |
|                                                                                                                                                                                                                            |
|   15:AGGREGATE (merge finalize)                                                                                                                                                                                            |
|   |  group by: <slot 23>, <slot 24>                                                                                                                                                                                        |
|   |  tuple ids: 7                                                                                                                                                                                                          |
|   |                                                                                                                                                                                                                        |
|   14:EXCHANGE                                                                                                                                                                                                              |
|      tuple ids: 7                                                                                                                                                                                                          |
|                                                                                                                                                                                                                            |
| PLAN FRAGMENT 5                                                                                                                                                                                                            |
|  OUTPUT EXPRS:                                                                                                                                                                                                             |
|   PARTITION: RANDOM                                                                                                                                                                                                        |
|                                                                                                                                                                                                                            |
|   STREAM DATA SINK                                                                                                                                                                                                         |
|     EXCHANGE ID: 14                                                                                                                                                                                                        |
|     HASH_PARTITIONED: <slot 23>, <slot 24>                                                                                                                                                                                 |
|                                                                                                                                                                                                                            |
|   5:AGGREGATE (update serialize)                                                                                                                                                                                           |
|   |  STREAMING                                                                                                                                                                                                             |
|   |  group by: `event_day`, `cuid`                                                                                                                                                                                         |
|   |  tuple ids: 7                                                                                                                                                                                                          |
|   |                                                                                                                                                                                                                        |
|   4:OlapScanNode                                                                                                                                                                                                           |
|      TABLE: visd_dws_topic_novel_wide_table_user                                                                                                                                                                           |
|      PREAGGREGATION: ON                                                                                                                                                                                                    |
|      PREDICATES: `sid` IN ('all'), `value` IN ('all'), `event_day` >= str_to_date('2019-04-29 00:00:00', '%Y-%m-%d %H:%i:%s'), `event_day` <= str_to_date('2019-05-06 00:00:00', '%Y-%m-%d %H:%i:%s'), `is_new` = 1.0      |
|      partitions=25/25                                                                                                                                                                                                      |
|      rollup: visd_dws_topic_novel_wide_table_user                                                                                                                                                                          |
|      buckets=800/800                                                                                                                                                                                                       |
|      cardinality=1060653874                                                                                                                                                                                                |
|      avgRowSize=29.075388                                                                                                                                                                                                  |
|      numNodes=30                                                                                                                                                                                                           |
|      tuple ids: 6                                                                                                                                                                                                          |
|                                                                                                                                                                                                                            |
| PLAN FRAGMENT 6                                                                                                                                                                                                            |
|  OUTPUT EXPRS:                                                                                                                                                                                                             |
|   PARTITION: RANDOM                                                                                                                                                                                                        |
|                                                                                                                                                                                                                            |
|   STREAM DATA SINK                                                                                                                                                                                                         |
|     EXCHANGE ID: 12                                                                                                                                                                                                        |
|     HASH_PARTITIONED: <slot 14>, <slot 15>                                                                                                                                                                                 |
|                                                                                                                                                                                                                            |
|   3:AGGREGATE (update serialize)                                                                                                                                                                                           |
|   |  STREAMING                                                                                                                                                                                                             |
|   |  group by: `event_day`, `cuid`                                                                                                                                                                                         |
|   |  tuple ids: 4                                                                                                                                                                                                          |
|   |                                                                                                                                                                                                                        |
|   2:OlapScanNode                                                                                                                                                                                                           |
|      TABLE: visd_dws_topic_novel_wide_table_user                                                                                                                                                                           |
|      PREAGGREGATION: ON                                                                                                                                                                                                    |
|      PREDICATES: `sid` IN ('all'), `value` IN ('all'), `event_day` >= str_to_date('2019-04-29 00:00:00', '%Y-%m-%d %H:%i:%s'), `event_day` <= str_to_date('2019-05-06 00:00:00', '%Y-%m-%d %H:%i:%s'), `is_read_dau` = 1.0 |
|      partitions=25/25                                                                                                                                                                                                      |
|      rollup: visd_dws_topic_novel_wide_table_user                                                                                                                                                                          |
|      buckets=800/800                                                                                                                                                                                                       |
|      cardinality=1060653874                                                                                                                                                                                                |
|      avgRowSize=29.079033                                                                                                                                                                                                  |
|      numNodes=30                                                                                                                                                                                                           |
|      tuple ids: 3                                                                                                                                                                                                          |
|                                                                                                                                                                                                                            |
| PLAN FRAGMENT 7                                                                                                                                                                                                            |
|  OUTPUT EXPRS:                                                                                                                                                                                                             |
|   PARTITION: RANDOM                                                                                                                                                                                                        |
|                                                                                                                                                                                                                            |
|   STREAM DATA SINK                                                                                                                                                                                                         |
|     EXCHANGE ID: 10                                                                                                                                                                                                        |
|     HASH_PARTITIONED: <slot 5>                                                                                                                                                                                             |
|                                                                                                                                                                                                                            |
|   1:AGGREGATE (update serialize)                                                                                                                                                                                           |
|   |  STREAMING                                                                                                                                                                                                             |
|   |  group by: `event_day`                                                                                                                                                                                                 |
|   |  tuple ids: 1                                                                                                                                                                                                          |
|   |                                                                                                                                                                                                                        |
|   0:OlapScanNode                                                                                                                                                                                                           |
|      TABLE: visd_dws_topic_novel_wide_table_user                                                                                                                                                                           |
|      PREAGGREGATION: ON                                                                                                                                                                                                    |
|      PREDICATES: `sid` IN ('all'), `value` IN ('all'), `event_day` >= str_to_date('2019-04-29 00:00:00', '%Y-%m-%d %H:%i:%s'), `event_day` <= str_to_date('2019-05-06 00:00:00', '%Y-%m-%d %H:%i:%s'), `is_dau` = 1.0      |
|      partitions=25/25                                                                                                                                                                                                      |
|      rollup: visd_dws_topic_novel_wide_table_user_no_cuid_userid_rollup                                                                                                                                                    |
|      buckets=800/800                                                                                                                                                                                                       |
|      cardinality=21101622                                                                                                                                                                                                  |
|      avgRowSize=170.06363                                                                                                                                                                                                  |
|      numNodes=30                                                                                                                                                                                                           |
|      tuple ids: 0                                                                                                                                                                                                          |
```
(2) wrong result:
```
+----------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------+
| Explain String                                                                                                                                                                                                             |
+----------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------+
| PLAN FRAGMENT 0                                                                                                                                                                                                            |
|  OUTPUT EXPRS:date(<slot 35>) | <slot 36> |                                                                                                                                                                                |
|   PARTITION: UNPARTITIONED                                                                                                                                                                                                 |
|                                                                                                                                                                                                                            |
|   RESULT SINK                                                                                                                                                                                                              |
|                                                                                                                                                                                                                            |
|   21:MERGING-EXCHANGE                                                                                                                                                                                                      |
|      limit: 1000                                                                                                                                                                                                           |
|      tuple ids: 13                                                                                                                                                                                                         |
|                                                                                                                                                                                                                            |
| PLAN FRAGMENT 1                                                                                                                                                                                                            |
|  OUTPUT EXPRS:                                                                                                                                                                                                             |
|   PARTITION: HASH_PARTITIONED: <slot 5>                                                                                                                                                                                    |
|                                                                                                                                                                                                                            |
|   STREAM DATA SINK                                                                                                                                                                                                         |
|     EXCHANGE ID: 21                                                                                                                                                                                                        |
|     UNPARTITIONED                                                                                                                                                                                                          |
|                                                                                                                                                                                                                            |
|   10:TOP-N                                                                                                                                                                                                                 |
|   |  order by: <slot 35> DESC                                                                                                                                                                                              |
|   |  offset: 0                                                                                                                                                                                                             |
|   |  limit: 1000                                                                                                                                                                                                           |
|   |  tuple ids: 13                                                                                                                                                                                                         |
|   |                                                                                                                                                                                                                        |
|   9:HASH JOIN                                                                                                                                                                                                              |
|   |  join op: LEFT OUTER JOIN (BROADCAST)                                                                                                                                                                                  |
|   |  hash predicates:                                                                                                                                                                                                      |
|   |    <slot 5> = <slot 29>                                                                                                                                                                                                |
|   |  tuple ids: 1 10N                                                                                                                                                                                                      |
|   |                                                                                                                                                                                                                        |
|   |----20:EXCHANGE                                                                                                                                                                                                         |
|   |       tuple ids: 10                                                                                                                                                                                                    |
|   |                                                                                                                                                                                                                        |
|   12:AGGREGATE (merge finalize)                                                                                                                                                                                            |
|   |  group by: <slot 5>                                                                                                                                                                                                    |
|   |  tuple ids: 1                                                                                                                                                                                                          |
|   |                                                                                                                                                                                                                        |
|   11:EXCHANGE                                                                                                                                                                                                              |
|      tuple ids: 1                                                                                                                                                                                                          |
|                                                                                                                                                                                                                            |
| PLAN FRAGMENT 2                                                                                                                                                                                                            |
|  OUTPUT EXPRS:                                                                                                                                                                                                             |
|   PARTITION: HASH_PARTITIONED: <slot 27>                                                                                                                                                                                   |
|                                                                                                                                                                                                                            |
|   STREAM DATA SINK                                                                                                                                                                                                         |
|     EXCHANGE ID: 20                                                                                                                                                                                                        |
|     UNPARTITIONED                                                                                                                                                                                                          |
|                                                                                                                                                                                                                            |
|   8:AGGREGATE (update finalize)                                                                                                                                                                                            |
|   |  output: count(<slot 28>)                                                                                                                                                                                              |
|   |  group by: <slot 27>                                                                                                                                                                                                   |
|   |  tuple ids: 10                                                                                                                                                                                                         |
|   |                                                                                                                                                                                                                        |
|   19:AGGREGATE (merge serialize)                                                                                                                                                                                           |
|   |  group by: <slot 27>, <slot 28>                                                                                                                                                                                        |
|   |  tuple ids: 9                                                                                                                                                                                                          |
|   |                                                                                                                                                                                                                        |
|   18:EXCHANGE                                                                                                                                                                                                              |
|      tuple ids: 9                                                                                                                                                                                                          |
|                                                                                                                                                                                                                            |
| PLAN FRAGMENT 3                                                                                                                                                                                                            |
|  OUTPUT EXPRS:                                                                                                                                                                                                             |
|   PARTITION: HASH_PARTITIONED: <slot 14>, <slot 15>                                                                                                                                                                        |
|                                                                                                                                                                                                                            |
|   STREAM DATA SINK                                                                                                                                                                                                         |
|     EXCHANGE ID: 18                                                                                                                                                                                                        |
|     HASH_PARTITIONED: <slot 27>                                                                                                                                                                                            |
|                                                                                                                                                                                                                            |
|   7:AGGREGATE (update serialize)                                                                                                                                                                                           |
|   |  STREAMING                                                                                                                                                                                                             |
|   |  group by: <slot 14>, <slot 15>                                                                                                                                                                                        |
|   |  tuple ids: 9                                                                                                                                                                                                          |
|   |                                                                                                                                                                                                                        |
|   6:HASH JOIN                                                                                                                                                                                                              |
|   |  join op: INNER JOIN (BROADCAST)                                                                                                                                                                                       |
|   |  hash predicates:                                                                                                                                                                                                      |
|   |    <slot 14> = <slot 23>                                                                                                                                                                                               |
|   |    <slot 15> = <slot 24>                                                                                                                                                                                               |
|   |  tuple ids: 4 7                                                                                                                                                                                                        |
|   |                                                                                                                                                                                                                        |
|   |----17:EXCHANGE                                                                                                                                                                                                         |
|   |       tuple ids: 7                                                                                                                                                                                                     |
|   |                                                                                                                                                                                                                        |
|   14:AGGREGATE (merge finalize)                                                                                                                                                                                            |
|   |  group by: <slot 14>, <slot 15>                                                                                                                                                                                        |
|   |  tuple ids: 4                                                                                                                                                                                                          |
|   |                                                                                                                                                                                                                        |
|   13:EXCHANGE                                                                                                                                                                                                              |
|      tuple ids: 4                                                                                                                                                                                                          |
|                                                                                                                                                                                                                            |
| PLAN FRAGMENT 4                                                                                                                                                                                                            |
|  OUTPUT EXPRS:                                                                                                                                                                                                             |
|   PARTITION: HASH_PARTITIONED: <slot 23>, <slot 24>                                                                                                                                                                        |
|                                                                                                                                                                                                                            |
|   STREAM DATA SINK                                                                                                                                                                                                         |
|     EXCHANGE ID: 17                                                                                                                                                                                                        |
|     UNPARTITIONED                                                                                                                                                                                                          |
|                                                                                                                                                                                                                            |
|   16:AGGREGATE (merge finalize)                                                                                                                                                                                            |
|   |  group by: <slot 23>, <slot 24>                                                                                                                                                                                        |
|   |  tuple ids: 7                                                                                                                                                                                                          |
|   |                                                                                                                                                                                                                        |
|   15:EXCHANGE                                                                                                                                                                                                              |
|      tuple ids: 7                                                                                                                                                                                                          |
|                                                                                                                                                                                                                            |
| PLAN FRAGMENT 5                                                                                                                                                                                                            |
|  OUTPUT EXPRS:                                                                                                                                                                                                             |
|   PARTITION: RANDOM                                                                                                                                                                                                        |
|                                                                                                                                                                                                                            |
|   STREAM DATA SINK                                                                                                                                                                                                         |
|     EXCHANGE ID: 15                                                                                                                                                                                                        |
|     HASH_PARTITIONED: <slot 23>, <slot 24>                                                                                                                                                                                 |
|                                                                                                                                                                                                                            |
|   5:AGGREGATE (update serialize)                                                                                                                                                                                           |
|   |  STREAMING                                                                                                                                                                                                             |
|   |  group by: `event_day`, `cuid`                                                                                                                                                                                         |
|   |  tuple ids: 7                                                                                                                                                                                                          |
|   |                                                                                                                                                                                                                        |
|   4:OlapScanNode                                                                                                                                                                                                           |
|      TABLE: visd_dws_topic_novel_wide_table_user                                                                                                                                                                           |
|      PREAGGREGATION: ON                                                                                                                                                                                                    |
|      PREDICATES: `sid` IN ('all'), `value` IN ('all'), `event_day` >= str_to_date('2019-04-29 00:00:00', '%Y-%m-%d %H:%i:%s'), `event_day` <= str_to_date('2019-05-06 00:00:00', '%Y-%m-%d %H:%i:%s'), `is_new` = 1.0      |
|      partitions=25/25                                                                                                                                                                                                      |
|      rollup: visd_dws_topic_novel_wide_table_user                                                                                                                                                                          |
|      buckets=800/800                                                                                                                                                                                                       |
|      cardinality=1060653874                                                                                                                                                                                                |
|      avgRowSize=29.077042                                                                                                                                                                                                  |
|      numNodes=30                                                                                                                                                                                                           |
|      tuple ids: 6                                                                                                                                                                                                          |
|                                                                                                                                                                                                                            |
| PLAN FRAGMENT 6                                                                                                                                                                                                            |
|  OUTPUT EXPRS:                                                                                                                                                                                                             |
|   PARTITION: RANDOM                                                                                                                                                                                                        |
|                                                                                                                                                                                                                            |
|   STREAM DATA SINK                                                                                                                                                                                                         |
|     EXCHANGE ID: 13                                                                                                                                                                                                        |
|     HASH_PARTITIONED: <slot 14>, <slot 15>                                                                                                                                                                                 |
|                                                                                                                                                                                                                            |
|   3:AGGREGATE (update serialize)                                                                                                                                                                                           |
|   |  STREAMING                                                                                                                                                                                                             |
|   |  group by: `event_day`, `cuid`                                                                                                                                                                                         |
|   |  tuple ids: 4                                                                                                                                                                                                          |
|   |                                                                                                                                                                                                                        |
|   2:OlapScanNode                                                                                                                                                                                                           |
|      TABLE: visd_dws_topic_novel_wide_table_user                                                                                                                                                                           |
|      PREAGGREGATION: ON                                                                                                                                                                                                    |
|      PREDICATES: `sid` IN ('all'), `value` IN ('all'), `event_day` >= str_to_date('2019-04-29 00:00:00', '%Y-%m-%d %H:%i:%s'), `event_day` <= str_to_date('2019-05-06 00:00:00', '%Y-%m-%d %H:%i:%s'), `is_read_dau` = 1.0 |
|      partitions=25/25                                                                                                                                                                                                      |
|      rollup: visd_dws_topic_novel_wide_table_user                                                                                                                                                                          |
|      buckets=800/800                                                                                                                                                                                                       |
|      cardinality=1060653874                                                                                                                                                                                                |
|      avgRowSize=29.078333                                                                                                                                                                                                  |
|      numNodes=30                                                                                                                                                                                                           |
|      tuple ids: 3                                                                                                                                                                                                          |
|                                                                                                                                                                                                                            |
| PLAN FRAGMENT 7                                                                                                                                                                                                            |
|  OUTPUT EXPRS:                                                                                                                                                                                                             |
|   PARTITION: RANDOM                                                                                                                                                                                                        |
|                                                                                                                                                                                                                            |
|   STREAM DATA SINK                                                                                                                                                                                                         |
|     EXCHANGE ID: 11                                                                                                                                                                                                        |
|     HASH_PARTITIONED: <slot 5>                                                                                                                                                                                             |
|                                                                                                                                                                                                                            |
|   1:AGGREGATE (update serialize)                                                                                                                                                                                           |
|   |  STREAMING                                                                                                                                                                                                             |
|   |  group by: `event_day`                                                                                                                                                                                                 |
|   |  tuple ids: 1                                                                                                                                                                                                          |
|   |                                                                                                                                                                                                                        |
|   0:OlapScanNode                                                                                                                                                                                                           |
|      TABLE: visd_dws_topic_novel_wide_table_user                                                                                                                                                                           |
|      PREAGGREGATION: ON                                                                                                                                                                                                    |
|      PREDICATES: `sid` IN ('all'), `value` IN ('all'), `event_day` >= str_to_date('2019-04-29 00:00:00', '%Y-%m-%d %H:%i:%s'), `event_day` <= str_to_date('2019-05-06 00:00:00', '%Y-%m-%d %H:%i:%s'), `is_dau` = 1.0      |
|      partitions=25/25                                                                                                                                                                                                      |
|      rollup: visd_dws_topic_novel_wide_table_user_no_cuid_userid_rollup                                                                                                                                                    |
|      buckets=800/800                                                                                                                                                                                                       |
|      cardinality=21090527                                                                                                                                                                                                  |
|      avgRowSize=170.1286                                                                                                                                                                                                   |
|      numNodes=30                                                                                                                                                                                                           |
|      tuple ids: 0                                                                                                                                                                                                          |
+----------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------+
```

2. remove order by event_day is ok:

**Comment by imay** [comment]
@qdore 

It seems that the problem is located in our order by implementation. We will look into it and try to reproduce this problem. 

And if you are convenient, you can help us to simplify this query and to reproduce this problem.

**Comment by qdore** [comment]
I'm very glad to help location this problem, I will try to simplify this query. ☺

**Comment by chenhao7253886** [comment]
Did you have upgraded to  our lastest version?  This seems to be solved before.

**Comment by qdore** [comment]
The version I tried is 0.9.0, I will try the newest master branch. And it will be nice if you paste me the patch commit log~

**Comment by chenhao7253886** [comment]
OK, i will conform it.

**Comment by lide-reed** [comment]
Can you add my weichat "leanderleereed"? I want to talk more detail about this issue?

**Comment by qdore** [comment]
Sure~

**Comment by lide-reed** [comment]
This bug has been fixed in the tag of Doris 0.9.0, and please upgrade your Doris.

**Comment by qdore** [comment]
Seems version 0.9.0-rc02  has solved this bug, Thx!
